README.md

```diff
@@ -29,95 +33,319 @@ are well established for RSE-ops, and which require further development. We can
+Installation of scientific software on HPC is a non-trivial task. To maximize the efficiency of any piece of software and maximally utilize an underlying resource, it's typically ideal to install it natively on the system, as opposed to a containerized environment, which is arguably a foundation of modern DevOps [^3].
+
+> Studies are showing that containerization overhead is really low -- so if you build the container
+> *for the system* the container performs just as well. But you lose portability doing that. There
+> are really interesting tradeoffs here. I want to have both portabilty and performacne.
+
+This means that HPC systems must exist with complex software stacks alongside one other with possibly conflicting dependencies.
+
+> usually kept separate and sort of sane with modules so that users can experiment.
+
+It means that software must be built across a wide set of architectures.
+
+> or at least optimized in a way that distros and container builders typcially do not do. This is
+> changing with x86_64 levels (new x86_64_v2, x86_64_v3, x86_64_v4 levels are in LLVM and GCC and
+> let you build more portable optimized binaries)
+
+For these reasons and typically smaller teams of linux administrators, there is little bandwidth left for testing,  benchmarking, or otherwise working on research software. Due to limited permissions, users are not always empowered to install their own software.
+
+> Our user install lots of software on their own in their home dirctories -- they build it or use
+> things like conda/pip/etc. I would be specific that they can't install and run services like they
+> can in the cloud and they really need to be able to do that.
+
+In practice, this means that linux administrators cannot devote any time to attempting to bring more DevOps-like practices to their system, and users don't have any control to do so either.
+
+> note that admins don't actually install the user-facing software on our machines -- user support
+> teams do. LC has ops (part of SAG), sysadmins (SAG), user support folks (DEG and some folks in
+> IMGS). Admins are doing low-level, OS-level stuff. User support is dealing with compilers,
+> programming environment, MPI, etc.
+
+A piece of software or new technology can only be embraced when it is asked for heavily by users, and this ask is finally heard by the linux administrators. Cloud native development gives users a lot more freedom.
+
+> this is tricky b/c cloud native also has a steeper learning curve than what our folks are used
+> to. You have to know how to administer a linux system -- or at least a container. So a lot more
```

Comment:
Hmm. I've had the experience of trying both for the first time, and I think HPC is much harder. When you have a web interface and lots of tutorials, it feels much friendlier than _just_ a command line with limited to no documentation (or documentation that is almost impossible to find and thus unlikely to find).

Reply:
Many people at the labs and in, like, the SIAM community don't agree with this.  They're used to their terminal and that's easy for them.  This might be a generation gap thing.  Some folks are also fine w/reading docs out of the repo -- I don't think it's the best thing my point is just that the difference here is subjective and depends on your background.  Cloud background is becoming more common but it's not a given.

Reply:
I guess I'm thinking forward to the newer generations.

Reply:
Maybe, or maybe this is also a bit biased by your own experience and ways of learning.  Cloud apps aren't "simple" or inherently easier to learn.  There are also great tutorials online for HPC -- see LLNL's HPC tutorials on MPI, OpenMP, etc., running jobs, etc.

I think maybe a better thing to say is that cloud apps are more widespread, that there is more material, and that can be advantageous for learning how to develop different types of apps.  It gets back to the point about how the mainstream is doing web development -- it's just easier to find documentation in many cases.